### PR TITLE
feat(web): include more options drop down in completed shares

### PR DESCRIPTION
**Pull request by tonypls**
## Issue
We're not tracking completed shares from the more options dropdown

## Description
This PR updates the useShare hook to include completed share tracking

### CODE REVIEW

File: web/src/features/panels/zone/ShareButton.tsx [hunk 1]
```diff
@@ -44,23 +44,16 @@ export function ShareButton({
...
-      ).then((shareCompleted) => {
-        if (shareCompleted) {
-          trackShareCompletion();
-        }
-      });
-    } else {
-      copyToClipboard(shareUrl, toastMessageCallback);
-    }
-    trackShareClick();
+    await (hasMobileUserAgent && (await CapShare.canShare())
```
Comment (cadeban) [inline web/src/features/panels/zone/ShareButton.tsx:47]: In my opinion this is more readable as an if statement. Now a reader is required to know the precedence of `await` vs the ternary operator. It's also a little strange to have a single await applied to multiple branches.

Snce we only care about the side effects here, I think an imperative statement more accurately describes the intent of the code.

File: web/src/utils/analytics.ts [hunk 2]
```diff
@@ -37,8 +37,10 @@ export enum ShareType {
   COMPLETED_SHARE = 'completed_share',
 }
 
-export const trackShare = (shareType: ShareType) => () =>
+export const trackShare = (shareType: ShareType) => () => {
+  console.log('share', shareType);
```
Comment (cadeban) [inline web/src/utils/analytics.ts:41]: nit: console.log here

File: web/src/hooks/useShare.ts [hunk 3]
```diff
@@ -21,7 +26,11 @@ export function useShare() {
   const share = useCallback(
     async (shareData: ShareOptions, callback?: (argument: string) => void) => {
       try {
+        trackShareClick();
         const result = await CapShare.share(shareData);
+        if (result) {
+          trackShareCompletion();
```
Comment (cadeban) [inline web/src/hooks/useShare.ts:32]: 🎉

File: web/src/utils/analytics.ts (hunk 2 above)
Comment (tonypls) [reply web/src/utils/analytics.ts]: Good spotting!

File: web/src/features/panels/zone/ShareButton.tsx (hunk 1 above)
Comment (tonypls) [reply web/src/features/panels/zone/ShareButton.tsx:47]: Same but our prettier rules force it to be ternary 🤦

Comment (cadeban) [reply web/src/features/panels/zone/ShareButton.tsx:47]: I should've known! 😧

Comment (cadeban) [reply web/src/features/panels/zone/ShareButton.tsx:47]: I'll approve this but I might look into changing Prettier when I have some down time...

**Review by cadeban — APPROVED** [review]

**Pull request merged**